### PR TITLE
Force Inline all the InterlockedAdd functions in stdlib

### PR DESCRIPTION
**Pull request by sriramm-nv**
Fixes issue #3969

This change forcibly inlines the InterlockedAdd functions when using byteAddress buffer.

The pattern seen when using nonUniformResourceInst on
    RWByteAddressBuffer
        buffer[NonUniformResourceIndex(uint(0))].InterlockedAdd(0, 1);

    is as follows:
    %ld = NonUniformResourceIndex(0)
    Call RWStructBufferInterlockedAdd(%ld, 0, 1)
This prevents any decoration from getting applied on the resource, since it is encapsulated by a function InterlockedAdd.

    where the InterlockedAdd expands to
    %gep = getElement(%buffer, 0)
    SpirvAsmInst(..., rwStructuredBufferGEP(%gep, 0), ...)

    Identified while debugging tests/spirv/coherent-2.slang

### CODE REVIEW

**Comment by sriramm-nv** [comment]
@csyonghe is this ok to merge?

File: source/slang/hlsl.meta.slang [hunk 1]
```diff
@@ -3116,6 +3120,7 @@ ${{{{
         }
     }
 
+    [ForceInline]
```
Comment (csyonghe) [inline source/slang/hlsl.meta.slang:3123]: There are more Interlocked functions that also need to be added. Like in line 3124, 3139, 3155, etc.

**Review by csyonghe — APPROVED** [review]

**Comment by jkwak-work** [comment]
Is there an issue number for this?
I am in favor of using ForceInline as much as we can.
But I am not sure what problem this is to address.
The description on the PR is not clear to me.
It says, "Identified while debugging tests/spirv/coherent-2.slang" but I am not sure what is identified.

**Comment by sriramm-nv** [comment]
> Is there an issue number for this? I am in favor of using ForceInline as much as we can. But I am not sure what problem this is to address. The description on the PR is not clear to me. It says, "Identified while debugging tests/spirv/coherent-2.slang" but I am not sure what is identified.

Created an issue #3969 for this change, and beyond.

**Pull request merged**